### ISSUE TITLE
Failing UI tests on iOS 26+ when building with Xcode 26.1 and Xcode 26.1

### ISSUE THREAD

**Issue by pylapp** [Orange-OpenSource/ouds-ios#1247]
# Context

## Before

Until the use of Xcode 26.0, everything worked fine with our UI tests.
For both simulators and real devices, for iOS 26+ et iOS < 26, it was possible to move across the app, find elements, interact with them and trigger any events to make our UI tests.
The accessibility inspector worked fine and was able to get the hierarchy of the app for both a real device and a simulator.
UI tests were ok both locally and on GitHub Actions workflows.

## After

However, with Xcode 26.1 (#1209) and Xcode 26.2 (#1246), almost all UI tests fail.
The accessibility inspector tool is not able to debug the view hierarchy for an app on a simulator or a real device.
The UI tests fail to find an element, with the same code base.
Nevertheless this is only the case for iOS 26+, not iOS < 26
UI tests failed both locally and on GitHub Actions workflows.

## Tests

- UI tests using Xcode 26.0 with simulator iOS 26+ = **pass** ✅
- UI tests using Xcode 26.0 with simulator iOS < 26 = **pass** ✅
- UI tests using Xcode 26.0 with device iOS 26+ = **pass** ✅
- UI tests using Xcode 26.0 with device iOS < 26 = **pass** ✅

- Accessibility Inspector, Xcode 26.0 with simulator iOS 26+ = **pass** ✅
- Accessibility Inspector, Xcode 26.0 with simulator iOS < 26 = **pass** ✅
- Accessibility Inspector, Xcode 26.0 with device iOS 26+ = **pass** ✅
- Accessibility Inspector, Xcode 26.0 with device iOS < 26 = **pass** ✅

- UI tests using Xcode 26.2 with simulator iOS 26+ = **fail** ❌
- UI tests using Xcode 26.2 with simulator iOS < 26 = **pass** ✅
- UI tests using Xcode 26.2 with device iOS 26+ = **fail** ❌
- UI tests using Xcode 26.2 with device iOS < 26 = **pass** ✅

- Accessibility Inspector, using Xcode 26.2 with simulator iOS 26+ = **fail** ❌
- Accessibility Inspector, using Xcode 26.2 with simulator iOS < 26 = **pass** ✅
- Accessibility Inspector, using Xcode 26.2 with device iOS 26+ = **fail** ❌
- Accessibility Inspector, using Xcode 26.2 with device iOS < 26 = **pass** ✅

# Logs

Some test function:
```swift
    @MainActor
    func testMakeScreenshotsForDocumentation_Button() {
        let app = launchApp()
        // Error on line below
        // 📄_ActionsDocumentationScreenshots.testMakeScreenshotsForDocumentation_Button()
        goToComponentsSheet(app)
        waitForButtonToAppear(withWording: "app_components_button_label", app)
        tapButton(withWording: "app_components_button_label", app)

        tapButton(withWording: "Strong", app)

        takeScreenshot(named: "component_button_", ACDC.buttonX, ACDC.buttonY, ACDC.buttonWidth, ACDC.buttonHeight, app)
    }
```

The helper:
```swift
    /// Opens the page of the components, i.e. tap on the 2nd of the tab bar
    @MainActor func goToComponentsSheet(_ app: XCUIApplication) {
        app.tabBars.buttons.element(boundBy: 1).tap()
    }
```

With the error on the only line of `goToComponentsSheet(_:)`
```text
.../ouds-ios-design-system-toolbox/DesignToolbox/DesignToolboxUITests/_/AppTestCase.swift:157 testMakeScreenshotsForDocumentation_Button(): Failed to tap Button (Element at index 1): No matches found for Descendants matching type TabBar from input {(
    Application, pid: 1883
)}
```

# Workaround (temporary)

- Run UI tests on simulator with iOS lower than 26
- Keep Xcode 26.2 because Xcode 26.0 and Xcode 26.1 not supported anymore in our internal FaaS

# Resources

- Apple Developer Forum publication: https://developer.apple.com/forums/thread/812307

# Definition of Done

- [ ] Test with new versions of Xcode until finding a working version
- [ ] Or find another solution for UI tests

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **1.1.0**
- Build number: **1768946674**
- Build tag: **6e5f834**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_
